docs/topics/properties.md

```diff
@@ -146,7 +146,26 @@ Such a property needs to fulfil the following requirements:
 * It must be initialized with a value of type `String` or a primitive type
 * It cannot be a custom getter
 
-Such properties can be used in annotations:
+The compiler will inline usages of the constant, but will not remove the field (this can be useful when using [reflection](reflection.md)). For example, this Kotlin code compiles to the following Java code:
```

Comment:
I suggest clarifying what "inlines usages of the constant" means -> "inline usage of the constant during compilation, replacing the reference to the constant with its actual value".